examples/scaffold.jl

```diff
@@ -0,0 +1,50 @@
+import MPI
+import NCCL
+using CuArrays
+using CUDAdrv
+using CUDAnative
+
+MPI.Init()
+comm   = MPI.COMM_WORLD
+myrank = MPI.Comm_rank(comm)
+nranks = MPI.Comm_size(comm)
+
+# Issues:
+# - Avoid allocations during allReduce
+
+print(stdout, ENV)
+
+@info "MPI initialized" myrank nranks
+
+if myrank == 0
+    uid = NCCL.UniqueID()
+else
+    uid = nothing
+end
+uid = MPI.bcast(uid, 0, comm)::NCCL.UniqueID
+
+dev = CuDevice(parse(Int, first(split(ENV["CUDA_VISIBLE_DEVICES"], ","))))
```

Comment:
The "right" way of doing this is:

```
    lcomm = MPI.Comm_split_type(mpicomm, MPI.MPI_COMM_TYPE_SHARED,
                                MPI.Comm_rank(mpicomm))
    CUDAnative.device!(MPI.Comm_rank(lcomm))
```
For NCCL to work best all devices need to be visible to all MPI processes and there should only be a maxium of n local ranks per n devices